src/services/services.ts

```diff
@@ -3010,21 +3010,30 @@ namespace ts {
             function tryGetGlobalSymbols(): boolean {

```

Comment:
I would call this tryGetContextualPropertyNameOrNameInScope. If that's too long, it's because the function is trying to do too many unrelated things.